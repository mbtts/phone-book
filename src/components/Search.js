import React, { PureComponent } from "react";

import PropTypes from "prop-types";

class Search extends PureComponent {
  render() {
    return (
      <input
        name="search"
        type="search"
        placeholder="Searchâ€¦"
        autoComplete="off"
        disabled={this.props.disabled}
        value={this.props.value}
        onChange={this.props.onChange}
      />
    );
  }
}

Search.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  disabled: PropTypes.bool.isRequired
};

export default Search;
