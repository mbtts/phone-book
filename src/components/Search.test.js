import React from "react";
import Search from "./Search";

describe("Search component", () => {
  it("should render initial state", () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <Search value="" disabled={true} onChange={onChange} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should invoke the change function when text entered", () => {
    const onChange = jest.fn();
    const name = "name";

    const wrapper = shallow(
      <Search value="" disabled={false} onChange={onChange} />
    );

    const event = {
      target: { name: "search", value: name }
    };
    wrapper.find("input").simulate("change", event);
    expect(onChange).toHaveBeenCalledWith(event);
  });
});
