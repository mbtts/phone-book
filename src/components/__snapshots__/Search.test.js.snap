// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should render initial state 1`] = `
<input
  autoComplete="off"
  className="search"
  disabled={true}
  name="search"
  onChange={[MockFunction]}
  placeholder="Searchâ€¦"
  type="search"
  value=""
/>
`;
